controllers/cluster/capabilities.go

```diff
@@ -0,0 +1,42 @@
+package cluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	v1 "github.com/openshift/api/config/v1"
+	clusterversion "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+)
+
+type Capabilities struct {
+	HasEtcdPDBQuorum, HasMachineAPI bool
+}
+
+// HasMachineAPICapability returns true if the cluster has the MachineAPI Enabled
+func HasMachineAPICapability(config *rest.Config) (bool, error) {
+	configV1Client, err := clusterversion.NewForConfig(config)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to create cluster version client")
+	}
+	cvs, err := configV1Client.ClusterVersions().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to check for Openshift cluster upgrade status")
```

Comment:
IIUC when upgrading we can expect more than one cluster version, but this error occurs when failing to fetch any cluster version.
Can you explain what's the reason for the syntax ?

Reply:
oops, that's a leftover from copy&paste I forgot to review before submitting :bow: